lxd/instance/drivers/driver_lxc.go

```diff
@@ -6321,6 +6321,58 @@ func (d *lxc) MigrateReceive(args instance.MigrateReceiveArgs) error {
 	}
 }
 
+// ConversionReceive establishes the filesystem connection, transfers the filesystem / block volume,
+// and creates an instance from it.
+func (d *lxc) ConversionReceive(args instance.ConversionReceiveArgs) error {
+	d.logger.Info("Conversion receive starting")
+	defer d.logger.Info("Conversion receive stopped")
+
+	// Wait for essential migration connections before negotiation.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	filesystemConn, err := args.FilesystemConn(ctx)
+	if err != nil {
+		return err
+	}
+
+	pool, err := storagePools.LoadByInstance(d.state, d)
+	if err != nil {
+		return err
+	}
+
+	// At this point we have already figured out the parent container's root
```

Comment:
Can we de-dupe this to avoid having it four times (convert and migrate for VMs and containers)